backend/app/nodes/llm/llm_utils.py

```diff
@@ -321,18 +414,45 @@ async def generate_text(
         "messages": messages,
         "temperature": temperature,
     }
-    if json_mode and not model_name.startswith("ollama"):
-        kwargs["response_format"] = {"type": "json_object"}
+
+    if json_mode:
+        if model_name.startswith("ollama"):
+            options = OllamaOptions(temperature=temperature, max_tokens=max_tokens)
+            response = await ollama_with_backoff(
+                model=model_name,
+                options=options,
+                messages=messages,
+                format="json",
+                api_base=api_base,
+            )
+        else:
+            # For both Azure and OpenAI, we need to set the response_format to json_object
+            kwargs["response_format"] = {"type": "json_object"}
+            # Add system message to enforce JSON mode
+            messages.insert(0, {
```

Comment:
We should (eventually) expose such hard-coded default messages to the end-user in the GUI so that they are fully aware of it and can still apply changes.